plugins/CustomVariables/CustomVariablesMetadataProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\CustomVariables;
+
+/**
+ * Provides information about installed custom variable slots, such as,
+ * the number of usable custom variables.
+ */
+class CustomVariablesMetadataProvider
+{
+    /**
+     * @var Model[]
+     */
+    private $logTableModels;
+
+    public function __construct(array $logTableModels)
+    {
+        $this->logTableModels = $logTableModels;
+    }
+
+    public function getNumUsableCustomVariables()
```

Comment:
I think there are also some tests for this methods. Maybe worth moving them to a new test case class but not needed. If so, only some of them. The ones that test the caching would be still needed in `CustomVariablesTest`